A27-xds-global-load-balancing.md

```diff
@@ -0,0 +1,426 @@
+          // The "config" field is optional; it may be missing if the
+          // credential type does not require config parameters.
+          "config": <JSON object containing config for the type>
+        }
+      ]
+    }
+  ],
+  "node": <JSON form of Node proto>
+}
+```
+
+Initially, the only type of channel creds we support will be
+`google_default`.  In the future, we will add a general-purpose
+mechanism for configuring arbitrary channel creds types with arbitrary
+configuration.
+
+The `node` field will be populated with the [xDS `Node`
+proto](https://github.com/envoyproxy/data-plane-api/blob/1adb5d54abb0e28ca409254d26fad1cf5535239b/envoy/api/v2/core/base.proto#L85).
+Note that the `build_version` field should not be specified in the
+bootstrap file, since that will be populated automatically based on the gRPC
+client's version.
+
+To allow for future changes in a backward-compatible way, unknown fields
+in the bootstrap file will be silently ignored.
+
+#### xds Resolver
+
+Clients will enable use of xDS by using the "xds" resolver in the
+target URI used to create the gRPC channel.  For example, a user may
+create a channel using the URI "xds:example.com:123" or
+"xds:///example.com:123", which will use xDS to establish contact with
+the server "example.com:123".  The "xds" URI scheme does not support any
+authority.
+
+When the channel attempts to connect, the xds resolver will
+instantiate an XdsClient object and use it to send an xDS request for
+a `Listener` resource with the name of the server that the client
+wishes to connect to (in the example above, "example.com:123").  If the
+resulting `Listener` does not include the `RouteConfiguration`,
+then the client will also send an xDS request for that resource.
+This information will be used to construct the gRPC [service
+config](https://github.com/grpc/grpc/blob/master/doc/service_config.md),
+which the resolver will return to the channel.
+
+In particular, the resulting service config will select use of the CDS
+LB policy, described below.  The configuration of the CDS policy will
+indicate which `Cluster` resource will be used.
+
+Note that the resolver will return an empty list of addresses, because
+in the xDS API flow, the addresses are not returned until the
+`ClusterLoadAssignment` resource is obtained later.
+
+The xds resolver will also return a reference to the XdsClient object
+that it instantiated.  This reference will be passed down to the LB
+policies, which is how the resolver and LB policies can share the same
+XdsClient object.
+
+#### CDS LB Policy
+
+The CDS LB policy will be the top-level LB policy.  Its configuration
+will be of the following form:
+
+```
+{
+  "cluster": "<cluster name>"
+}
+```
+
+The CDS policy will use the XdsClient object passed in from the xds resolver
+to query for the `Cluster` resource with the cluster name from the LB
+policy configuration.
+
+Initially, the CDS policy will support only hard-coded cluster names, so
```

Comment:
Do you mean "policy names" instead of "cluster names"?

Reply:
No, it was correct as written, but the wording was a bit confusing.  I've attempted to clarify.